lib/ansible/plugins/connection/ssh.py

```diff
@@ -906,6 +858,43 @@ def _examine_output(self, source: str, state: str, b_chunk: bytes, sudoable: boo
 
         return b''.join(output), remainder
 
+    def _init_shm(self) -> tuple[dict[str, str], dict[str, t.Any]]:
+        env = os.environ.copy()
+        popen_kwargs: dict[str, t.Any] = {}
+
+        conn_password = self.get_option('password') or self._play_context.password
+        pkcs11_provider = self.get_option("pkcs11_provider")
+        if not conn_password and pkcs11_provider:
+            raise AnsibleError("to use pkcs11_provider you must specify a password/pin")
+
+        if not conn_password:
+            return env, popen_kwargs
+
+        b_conn_password = conn_password.encode('utf-8')
```

Comment:
It seems like there should be a definitive "I won't be using a password" signal somewhere that could explicitly prevent lighting this up.